base/base64.jl

```diff
@@ -134,4 +165,48 @@ base64(x...) = base64(write, x...)
 
 #############################################################################
 
+type Base64DecodePipe <: IO
+    io::IO
+    # reading works in blocks of 4 characters that are decoded into 3 bytes and 2 of them cached
+    cache::Vector{UInt8}
+    encvec::Vector{UInt8}
+
+    function Base64DecodePipe(io::IO)
+        b = new(io,[],[])
+        finalizer(b, close)
+        return b
+    end
+end
+
+function read(b::Base64DecodePipe, t::Type{UInt8})
+    if length(b.cache) > 0
+        val = shift!(b.cache)
+    else
+        empty!(b.encvec)
+        while !eof(b.io) && length(b.encvec) < 4
+            c::UInt8 = read(b.io, t)
+            @inbounds if revb64chars[c] != sentinel
+                push!(b.encvec, c)
+            end
+        end
+        val = b64decode!(b.encvec,b.cache)
+    end
+    val
+end
+
+function eof(b::Base64DecodePipe)
+    return length(b.cache) == 0 && eof(b.io)
+end
+
+function close(b::Base64DecodePipe)
+end
+
+# Decodes a base64-encoded string
+function base64decode(s)
+    b = IOBuffer(s)
+    decoded = readall(Base64DecodePipe(b))
```

Comment:
I think this is wrong.  It should be `readbytes`, not `readall`.  The Base64-encoded data need not produce valid UTF-8 string.